### PR TITLE
Fix julia 0.7 deprecations, min req. set to 0.7-

### PR DESCRIPTION
**Pull request by lucatrv**

### CODE REVIEW

**Comment by ScottPJones** [comment]
Instead of setting the minimum for Julia to 0.7-, could it update the version of Compat used?

File: src/ArgParse.jl [hunk 1]
```diff
@@ -1937,10 +1937,10 @@ function preparse(c::Channel, state::ParserState, settings::ArgParseSettings)
             state.arg_delim_found = true
             state.token = nothing
             state.token_arg = nothing
-            shift!(args_list)
+            popfirst!(args_list)
             continue
         elseif startswith(arg, "--")
-            eq = search(arg, '=')
+            eq = findnext('=', arg)
             if eq != 0
```
Comment (jbdaley) [inline src/ArgParse.jl]: findnext returns nothing if the pattern is not found.

```
+            eq = findfirst(x -> isequal(x, '='), arg)
-            if eq != 0
+            if eq !== nothing
```

Comment (lucatrv) [reply src/ArgParse.jl]: I replaced `eq = findnext('=', arg)` with `eq = findfirst(equalto('='), arg)`.

**Comment by carlobaldassi** [comment]
Sorry for the delay and thanks for the PR. However, since a few more changes have intervened in the meanwhile and I wanted to keep support for julia 0.6, I have now enclosed these changes in another commit (cdf20c8c7dff54cf594fc619280967340df9af8f).

**Pull request closed without merging**